code/game/turfs/simulated/wall_attacks.dm

```diff
@@ -69,27 +68,44 @@
 
 	try_touch(user, rotting)
 
-/turf/simulated/wall/attack_generic(var/mob/user, var/damage, var/attack_message, var/wallbreaker)
+/turf/simulated/wall/attack_generic(mob/M, damage, attack_message)
+	log_and_message_admins("attacked [src]: [jumplink(src)]")
+	M.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
 
-	radiate()
-	if(!istype(user))
-		return
+	if(locate(/obj/effect/overlay/wallrot) in src)
+		return success_smash(M)
 
-	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
-	var/rotting = (locate(/obj/effect/overlay/wallrot) in src)
-	if(!damage || !wallbreaker)
-		try_touch(user, rotting)
+	radiate()
+	if(!istype(M))
 		return
```

Comment:
Since you're checking for `M`'s type, maybe do that before doing other manipulations with it?